### PR TITLE
Decompile scrcmd_18 -> scrcmd_party

### PR DESCRIPTION
**Pull request by tgsm**

### CODE REVIEW

**Comment by tgsm** [comment]
Force-push documents a variable I forgot to name before pushing, whoops

File: arm9/src/scrcmd_party.c [hunk 1]
```diff
@@ -0,0 +1,813 @@
...
+    {
+        friendship = 0xFF;
+    }
+
+    SetMonData(party_mon, MON_DATA_FRIENDSHIP, &friendship);
+
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_DepletePartyMonFriendship(struct ScriptContext* ctx)
```
Comment (red031000) [inline arm9/src/scrcmd_party.c:346]: SubtractPartyMonFriendship (or RemovePartyMonFriendship) would be preferable here imo

File: arm9/src/scrcmd_party.c [hunk 2]
```diff
@@ -0,0 +1,813 @@
...
+                }
+            }
+        }
+    }
+
+    *ret_ptr = (u16)available_mons;
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_CountAvailablePartyMonsAndPCMons(struct ScriptContext* ctx)
```
Comment (red031000) [inline arm9/src/scrcmd_party.c:474]: CountAvailablePartyAndPCMons would flow better imo

File: arm9/src/scrcmd_party.c [hunk 3]
```diff
@@ -0,0 +1,813 @@
...
+    party = SavArray_PlayerParty_get(ctx->unk80->saveBlock2);
+
+    for (ribbon_idx = 0, ribbons = 0; ribbon_idx < 80; ribbon_idx++)
+    {
+        for (u16 i = 0; i < party_count; i++)
+        {
+            struct Pokemon* party_mon = GetPartyMonByIndex(party, i);
+
+            BOOL party_mon_is_egg = (BOOL)GetMonData(party_mon, MON_DATA_IS_EGG, NULL);
+            if (party_mon_is_egg)
```
Comment (red031000) [inline arm9/src/scrcmd_party.c:734]: honestly this feels like a bug, it'll only count mons before the first egg

File: arm9/src/scrcmd_party.c [hunk 4]
```diff
@@ -0,0 +1,813 @@
...
+        {
+            ribbons++;
+        }
+    }
+
+    *ret_ptr = ribbons;
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_Unk022F(struct ScriptContext* ctx)
```
Comment (red031000) [inline arm9/src/scrcmd_party.c:717]: I assume this func is supposed to get the total number of types of ribbons held by the party

**Review by red031000 — CHANGES_REQUESTED** [review]

**Review by red031000 — APPROVED** [review]

**Pull request merged**